Upgrading to 3.1.300 SDK breaks Artifact uploads on Windows cloud builds

I upgraded to Visual Studio 16.6 which forced an upgrade to the 3.6.0 compiler build tools:

<PackageReference Include="Microsoft.Net.Compilers.Toolset" Version="3.6.0" PrivateAssets="all" />

and global.json upgrade for sdk to 3.1.300.  Builds fine locally but fails in GitHub and Azure Windows build clients.

I tracked down the cause to the 3.6.0 build tools producing the following file in the obj folder:
.NETStandard,Version=v2.0.AssemblyAttributes.cs

The artifact publishing scripts on Windows are failing due to the comma after .NetStandard.  In the Publish artifacts stage the following error is produced:

```
2020-05-21T16:19:24.2420521Z ##[error]cmd : The syntax of the command is incorrect.
At C:\AzureAgent\_work\1\s\azure-pipelines\artifacts\_stage_all.ps1:36 char:9
+         cmd /c mklink $Link $Target | Out-Null
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~

    + CategoryInfo          : NotSpecified: (The syntax of the command is incorrect.:String) [], RemoteException

    + FullyQualifiedErrorId : NativeCommandError
2020-05-21T16:19:24.2448131Z ##[error]PowerShell exited with code '1'.
```

I have verified the breaking change is the inclusion of new file with a comma in the 3.6.0 compiler toolset.

There is no bug in the template until you upgrade to Visual Studio 16.6.  Upgrading to Visual Studio 16.6 forced me to upgrade global.json sdk version to 3.1.300 and compiler build tools to 3.6.  The later change create a new file in the output directory: .NETStandard,Version=v2.0.AssemblyAttributes.cs which contains a comma.  The template script mentioned above calls mklink on Windows which fails to link an artifact from the Release to the coverageResults-Windows folder. 

I assuming someday consumers of the template will upgrade to Visual Studio 16.6 and be in the same situation.  I don't think this is a bug against AzP or GitHub but a bug in the script that handles the linking of coverage files.

```
Linking C:\AzureAgent\_work\1\a\coverageResults-Windows\obj\PureActive.Core\Release\netstandard2.0\.NETStandard,Version=v2.0.AssemblyAttributes.cs to C:\AzureAgent\_work\1\s\obj\PureActive.Core\Release\netstandard2.0\.NETStandard,Version=v2.0.AssemblyAttributes.cs in C:\AzureAgent\_work\1\a\coverageResults-Windows\obj\PureActive.Core\Release\netstandard2.0
##[error]cmd : The syntax of the command is incorrect.
At C:\AzureAgent\_work\1\s\azure-pipelines\artifacts\_stage_all.ps1:36 char:9
+         cmd /c mklink "$Link" "$Target" | Out-Null
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

    + CategoryInfo          : NotSpecified: (The syntax of the command is incorrect.:String) [], RemoteException

    + FullyQualifiedErrorId : NativeCommandError
```

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -24,11 +24,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.9.8" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.0.0" PrivateAssets="all" />
     <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0" PrivateAssets="All" />
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
-    <PackageReference Include="Nerdbank.GitVersioning" Version="3.1.74" PrivateAssets="all" />
-    <PackageReference Include="Microsoft.Net.Compilers.Toolset" Version="3.5.0" PrivateAssets="all" />
+    <PackageReference Include="Nerdbank.GitVersioning" Version="3.1.91" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.Net.Compilers.Toolset" Version="3.6.0" PrivateAssets="all" />
     <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.164" PrivateAssets="all" />
   </ItemGroup>
 

--- a/azure-pipelines/artifacts/_stage_all.ps1
+++ b/azure-pipelines/artifacts/_stage_all.ps1
@@ -33,7 +33,7 @@ function Create-SymbolicLink {
     if ($IsMacOS -or $IsLinux) {
         ln $Target $Link | Out-Null
     } else {
-        cmd /c mklink $Link $Target | Out-Null
+        cmd /c "mklink `"$Link`" `"$Target`"" | Out-Null
     }
 }
 

--- a/global.json
+++ b/global.json
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "3.1.201",
+    "version": "3.1.300",
     "rollForward": "patch",
     "allowPrerelease": false
   }

--- a/test/Library.Tests/Library.Tests.csproj
+++ b/test/Library.Tests/Library.Tests.csproj
@@ -6,7 +6,7 @@
 
   <ItemGroup>
     <PackageReference Include="coverlet.msbuild" Version="2.8.1" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.6.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.6.1" />
     <PackageReference Include="xunit" Version="2.4.1" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.4.1" />
   </ItemGroup>

